src/test/java/io/confluent/kafkarest/unit/ConsumerManagerTest.java

```diff
@@ -172,38 +177,50 @@ public void testConsumerNormalOps() throws InterruptedException, ExecutionExcept
     String cid = consumerManager.createConsumer(
         groupName, new ConsumerInstanceConfig(EmbeddedFormat.BINARY));
     sawCallback = false;
+    actualException = null;
```

Comment:
@gawth It should throw an `AssertionError` where the cause is the exception that occurred in the other thread: https://github.com/apache/kafka/blob/trunk/connect/runtime/src/test/java/org/apache/kafka/connect/util/TestBackgroundThreadExceptionHandler.java#L35 That should mean anything that logs the exception stacktrace should also provide the original exception's stack trace, which should make it easy to track down the source of the issue. You do have to scan far enough in the message to see the second stacktrace, but we could also just add a message, something like `"A background thread threw an exception during the test, see the cause exception for details."` to make it obvious what happened.